### PR TITLE
The cgroup name can't be nil while systemReserved is designated

### LINKED ISSUE #83228: The cgroup name can't be nil while systemReserved is designated
**What happened**:
Currently kubelet just check 'systemReserved' and 'kubeReserved' parameters, did not check cgourp name. Or:
\----
Sep 27 14:38:56 k8s-0454-db511213 kubelet[2728]: F0927 14:38:56.247087    2728 kubelet.go:1359] Failed to start ContainerManager Failed to enforce System Reserved Cgroup Limits on "": failed to set supported cgroup subsystems for cgroup []: Failed to set config for supported subsystems : failed to write 536870912 to memory.limit_in_bytes: write /sys/fs/cgroup/memory/memory.limit_in_bytes: invalid argument

\----

**What you expected to happen**:
when system-reserved is enabled on enforceNodeAllocatable, both systemReserved and systemReservedCgroup should be designated, and
when kube-reserved is enabled on enforceNodeAllocatable, both kubeReserved and kubeReservedCgroup should be designated.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by FengyunPan2**
related to #83228
Currently kubelet just check 'systemReserved' and 'kubeReserved'
parameters, did not check cgourp name. This pr would check cgroup
name before enforcing System reserved on cgroup system.slice

/kind bug

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/cm/node_container_manager_linux.go [hunk 1]
```diff
@@ -102,34 +102,47 @@ func (cm *containerManagerImpl) enforceNodeAllocatableCgroups() error {
 	}
 	// Now apply kube reserved and system reserved limits if required.
 	if nc.EnforceNodeAllocatable.Has(kubetypes.SystemReservedEnforcementKey) {
-		klog.V(2).Infof("Enforcing System reserved on cgroup %q with limits: %+v", nc.SystemReservedCgroupName, nc.SystemReserved)
-		if err := enforceExistingCgroup(cm.cgroupManager, cm.cgroupManager.CgroupName(nc.SystemReservedCgroupName), nc.SystemReserved); err != nil {
-			message := fmt.Sprintf("Failed to enforce System Reserved Cgroup Limits on %q: %v", nc.SystemReservedCgroupName, err)
-			cm.recorder.Event(nodeRef, v1.EventTypeWarning, events.FailedNodeAllocatableEnforcement, message)
-			return fmt.Errorf(message)
+		if len(nc.SystemReservedCgroupName) == 0 {
+			klog.Warningf("Warning: when %s is enabled on enforceNodeAllocatable, both systemReserved and systemReservedCgroup should be designated,"+
```
Comment (tedyu) [inline pkg/kubelet/cm/node_container_manager_linux.go]: I think an error should be returned here.

Comment (FengyunPan2) [reply pkg/kubelet/cm/node_container_manager_linux.go]: Ok, update it now.

File: pkg/kubelet/cm/node_container_manager_linux.go [hunk 2]
```diff
@@ -111,6 +118,13 @@ func (cm *containerManagerImpl) enforceNodeAllocatableCgroups() error {
 		cm.recorder.Eventf(nodeRef, v1.EventTypeNormal, events.SuccessfulNodeAllocatableEnforcement, "Updated limits on system reserved cgroup %v", nc.SystemReservedCgroupName)
 	}
 	if nc.EnforceNodeAllocatable.Has(kubetypes.KubeReservedEnforcementKey) {
+		if len(nc.KubeReservedCgroupName) == 0 {
```
Comment (tedyu) [inline pkg/kubelet/cm/node_container_manager_linux.go]: Maybe extract the body of if block to a method so that it can be used for line 105 as well.

Comment (FengyunPan2) [reply pkg/kubelet/cm/node_container_manager_linux.go]: Done

**Comment by FengyunPan2** [comment]
@neolit123 Yes, this PR is useful for users, introduce a user-facing change.

**Comment by neolit123** [comment]
> Yes, this PR is useful for users, introduce a user-facing change.

please add the note.
see an example here https://github.com/kubernetes/kubernetes/pull/83694

**Pull request closed without merging**